### PR TITLE
Add fetch to retrieve data from backend

### PR DESCRIPTION
**Pull request by femioladeji**
# JIRA TICKET NAME:

## SUMMARY:

Before you submit this PR:
- Ensure the pull request you are about to create, is for an existing JIRA ticket
- Make sure the branch name is the same as the ticket name
- Make sure the title of the PR contains the ticket's JIRA ID and link
- Pull against master to get the latest changes
- Make sure your tests pass
- Include screenshots if necessary but do so on the JIRA page not here.
- Include a meaningful summary
- Add the pull request link to the issue after creation
- Remember to delete these directions. they should not be in your PR.
- Avoid multiple PRs for the same issue

### CODE REVIEW

File: app/js/components/tabs/tabsComponent.jsx [hunk 1]
```diff
@@ -30,19 +31,35 @@ class TabsComponent extends Component {
...
+        const getData = new Promise(function(resolve, reject) {
+            apiHelper.get(url).then(response => {
+                console.log(response);
+                response.json().then(data => {
+                    resolve(data);
+                });
+            });
+        });
+        return getData;
+        // ApiHelper.get(url);
```
Comment (danielonodje) [inline app/js/components/tabs/tabsComponent.jsx:45]: can you remove this comment?

**Comment by dkayiwa** [comment]
Can you always include the ticket number in the commit message.
You also need to specify which data you are fetching, in your commit and pull request.

File: app/js/components/tabs/tabsComponent.jsx [hunk 2]
```diff
@@ -30,19 +31,35 @@ class TabsComponent extends Component {
...
+            );
+        });
     }
 
     render(){
         return (
             <div className="col-sm-12 section">
-                <TabBarComponent tabs={this.state.tabs} drawTabHeader={this.drawTabHeader}/>
-                <TabContentComponent tabs={this.state.tabs} drawComponent={this.drawComponent}/>
+                <TabBarComponent tabs={this.state.tabs} drawTabHeader={this.drawTabHeader} />
```
Comment (danielonodje) [inline app/js/components/tabs/tabsComponent.jsx:61]: there's a space that's not needed in this line

File: app/js/helpers/apiHelper.js [hunk 3]
```diff
@@ -24,12 +26,13 @@ export class ApiHelper {
...
   send() {
-    const response = this.requestLibrary(this.requestUrl, this.requestOptions)
+    const request = this.requestLibrary;
+    const response = request(`${BASE_URL}${this.requestUrl}`, this.requestOptions)
       .then((data) => {
         return this.mocked ? data : data.json();
       })
       .catch((error) => {
-        return error.json;
+        // return error.json;
```
Comment (femioladeji) [inline app/js/helpers/apiHelper.js:35]: can we conclude on how we want to be handling errors?

Comment (danielonodje) [inline app/js/helpers/apiHelper.js:35]: why's this commented out?

File: package.json [hunk 4]
```diff
@@ -17,6 +17,7 @@
     "redux": "^3.6.0",
     "redux-promise-middleware": "^4.2.0",
     "redux-thunk": "^2.1.0",
+    "shortid": "^2.2.6",
```
Comment (danielonodje) [inline package.json:20]: what does this component do?

**Review by danielonodje — APPROVED** [review]
Thanks a lot man. you've made our work easier.

File: app/js/helpers/apiHelper.js (hunk 3 above)
Comment (emekanwankwo) [inline app/js/helpers/apiHelper.js:35]: You may want to replace the commented code with an exception handler.

File: app/js/components/tabs/tabsComponent.jsx (hunk 1 above)
Comment (emekanwankwo) [inline app/js/components/tabs/tabsComponent.jsx:45]: You can remove the commented code.

**Comment by femioladeji** [comment]
Hi @dkayiwa I actually made a mistake, I wanted to raise the PR to the forked repository (not upstream) so that we could review it together. I'll have to close this. I'm sorry about that

File: package.json (hunk 4 above)
Comment (fortunee) [reply package.json:20]: Component or module? @andela-odaniel
I think it generates unique IDs that can be used as keys, is that right?@andela-foladeji

**Pull request closed without merging**